Ci fixes

bgeltz opened the pull request:
A collection of changes to get our CI pipeline working with the service branch.  The build process and dependency resolution has changed a bit, and there were many changes needed to adhere to the architecture on the service branch.

bgeltz committed:
CI fixes to the GitHub actions workflow

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo committed:
Fix copying_headers/test-dist so it uses both distribution tarballs

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Use OMP 11

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo commented on .github/workflows/build.yml:
```diff
@@ -68,37 +67,51 @@ jobs:
+      run: sudo apt-get update && sudo apt-get install libelf-dev mpich libmpich-dev libomp-9-dev libsystemd-dev libgirepository1.0-dev
+    - name: install base and service dir python dependencies
+      run: |
+           python3 -m pip install --upgrade pip setuptools wheel pep517
```
I guess we could go back to the standard github script now that the requirements.txt no longer reference geopmdpy?

bgeltz replied:
No, we can't. The PyGobject requirement is problematic, and requires pip517 in order for pip to be able to successfully complete the source build within the GH actions VM.  There's no option with the standard github script thing to use pep517.

Example error:
https://github.com/bgeltz/geopm/runs/3591856424?check_suite_focus=true

You can disable pep517 outright:
https://github.com/bgeltz/geopm/runs/3591889886?check_suite_focus=true

But it seems better to try and use it since it's the new standard.

cmcantalupo commented on service/geopmdpy/__init__.py:
```diff
@@ -34,6 +34,7 @@
 
 """
 
+from __future__ import absolute_import
```
This is only for supporting python 2.x, we really want to get rid of all of these abolute_import lines, not add them back

bgeltz replied:
Should I do this a part of this PR, or a new one?

cmcantalupo commented on service/geopmdpy/runtime.py:
```diff
@@ -35,6 +35,8 @@
 """
 
+from __future__ import absolute_import
+from __future__ import division
```
Similar issue with division, we are in the __future__ :)

bgeltz replied:
Sure, just trying to satisfy the test we have that checks all this stuff.  I can rip it all out in this PR too if that's the desire.

cmcantalupo reviewed (changes requested):
Just a couple comments

cmcantalupo committed:
Compile service with GNU toolchain in build.sh

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Remove dependencies from io.github.geopm.xml

- The dependecy is only on the __doc__ strings but adding the dependency
  requires libgeopmd.so.
- This file is required for "make dist" so it should not have a binary
  requirement.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz committed:
Bring back test-dist

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
